src/libostree/ostree-kernel-args.c

```diff
@@ -848,3 +848,26 @@ ostree_kernel_args_contains (OstreeKernelArgs  *kargs,
 
   return g_hash_table_contains (kargs->table, key);
 }
+
+/**
+ * ostree_kernel_args_delete_if_present:
+ * @kargs: a OstreeKernelArgs instance
+ * @arg: key or key/value pair to be added
+ * @error: an GError instance
+ *
+ * Deletes @arg which is in the form of key=value pair from the hash table kargs->table.
+ *
+ * Returns: %TRUE on success, %FALSE on failure
+ *
+ * Since: 2022.5
+ **/
+gboolean
+ostree_kernel_args_delete_if_present (OstreeKernelArgs  *kargs,
+                                      const char        *arg,
+                                      GError           **error)
+{
+  // Don't insert a duplicate key.
+  if (ostree_kernel_args_contains (kargs, arg))
+    ostree_kernel_args_delete (kargs, arg, error);
```

Comment:
The comment is wrong and this should return the value from `ostree_kernel_args_delete`. The other way to implement this would be to catch the error from `ostree_kernel_args_delete`, but it uses the generic `G_IO_ERROR_FAILED`, so you can't really tell the difference between non-existent key and some other error.